sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -910,85 +910,49 @@ class AdaptiveQueryExecSuite
     }
   }
 
-  test("SPARK-47148: AQE should avoid to materialize ShuffleQueryStage on the cancellation") {
+  test("SPARK-47148: AQE should avoid to submit shuffle job on cancellation") {
```

Comment:
This test case seems to be a bit unstable:
- https://github.com/apache/spark/actions/runs/11126022068/job/30915049564
- https://github.com/apache/spark/actions/runs/11137584720/job/31007122056
```
[info] - SPARK-47148: AQE should avoid to submit shuffle job on cancellation *** FAILED *** (6 seconds, 94 milliseconds)
[info]   "Multiple failures in stage materialization." did not contain "coalesce test error" (AdaptiveQueryExecSuite.scala:939)
```
- https://github.com/apache/spark/actions/runs/11162559129/job/31027480672
```
- SPARK-47148: AQE should avoid to submit shuffle job on cancellation *** FAILED ***
  "[SCALAR_SUBQUERY_TOO_MANY_ROWS] More than one row returned by a subquery used as an expression. SQLSTATE: 21000
  == SQL (line 1, position 12) ==
  SELECT id, (SELECT slow_udf() FROM range(2)) FROM range(5)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  " did not contain "coalesce test error" (AdaptiveQueryExecSuite.scala:939)
```

any good solutions?

Reply:
I think it's because the slow_udf is not slow enough and the shuffle stage was submitted too early. Can you try to increase the sleep time in slow_udf and see if it fixes the problem?